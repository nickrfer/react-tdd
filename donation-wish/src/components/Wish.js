import React, { Component } from 'react';
import { Form, FormGroup, FormControl, ControlLabel, Button } from 'react-bootstrap';

class Wish extends Component {
  constructor() {
    super();

    this.state = { person: '', wish: '' };
  }

  render() {
    return (
      <div className='wish'>
        <Form>
          <FormGroup>
            <ControlLabel>Person</ControlLabel>
            <FormControl
              className='input-person'
              onChange={event => this.setState({ person: event.target.value })}
            />
          </FormGroup>
          <FormGroup>
            <ControlLabel>Item</ControlLabel>
            <FormControl
              className='input-item'
              onChange={event => this.setState({ item: event.target.value })}
            />
          </FormGroup>
        </Form>
        <Button
          className='btn-remove'
          onClick={() => this.props.removeWish(this.props.wish.id)}
        >
          Remove Wish
        </Button>
      </div>
    );
  }
}

export default Wish;
