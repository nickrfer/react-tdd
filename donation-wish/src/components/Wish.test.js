import React from 'react';
import { shallow } from 'enzyme';
import Wish from './Wish';

describe('Wish', () => {
  const mockRemove = jest.fn();
  const id = 1;
  const props = { wish: { id }, removeWish: mockRemove };
  const wish = shallow(<Wish {...props} />);

  it('renders properly', () => {
    expect(wish).toMatchSnapshot();
  });

  it('initializes a person and wish in `state`', () => {
    expect(wish.state()).toEqual({ person: '', wish: '' });
  });

  describe('when typing into the person input', () => {
    const person = 'Uncle';

    beforeEach(() => {
      wish.find('.input-person').simulate('change', { target: { value: person } });
    });

    it('updates the person in `state`', () => {
      expect(wish.state().person).toEqual(person);
    })
  });

  describe('when typing into the present input', () => {
    const item = 'Old 3ds';

    beforeEach(() => {
      wish.find('.input-item').simulate('change', { target: { value: item } });
    });

    it('updates the present in `state`', () => {
      expect(wish.state().item).toEqual(item);
    });
  });

  describe('when click the `Remove Wish` button', () => {
    beforeEach(() => {
      wish.find('.btn-remove').simulate('click');
    });

    it('calls the removeWish callback', () => {
      expect(mockRemove).toHaveBeenCalledWith(id);
    });
  })
});
