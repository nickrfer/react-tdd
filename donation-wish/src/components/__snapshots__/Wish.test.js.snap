// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wish renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wish
    removeWish={[Function]}
    wish={
        Object {
            "id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
>
          <FormGroup
                    bsClass="form-group"
          >
                    <ControlLabel
                              bsClass="control-label"
                              srOnly={false}
                    >
                              Person
                    </ControlLabel>
                    <FormControl
                              bsClass="form-control"
                              className="input-person"
                              componentClass="input"
                              onChange={[Function]}
                    />
          </FormGroup>
          <FormGroup
                    bsClass="form-group"
          >
                    <ControlLabel
                              bsClass="control-label"
                              srOnly={false}
                    >
                              Item
                    </ControlLabel>
                    <FormControl
                              bsClass="form-control"
                              className="input-item"
                              componentClass="input"
                              onChange={[Function]}
                    />
          </FormGroup>
</Form>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="btn-remove"
          disabled={false}
          onClick={[Function]}
>
          Remove Wish
</Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": Array [
            <FormGroup
              bsClass="form-group"
>
              <ControlLabel
                            bsClass="control-label"
                            srOnly={false}
              >
                            Person
              </ControlLabel>
              <FormControl
                            bsClass="form-control"
                            className="input-person"
                            componentClass="input"
                            onChange={[Function]}
              />
</FormGroup>,
            <FormGroup
              bsClass="form-group"
>
              <ControlLabel
                            bsClass="control-label"
                            srOnly={false}
              >
                            Item
              </ControlLabel>
              <FormControl
                            bsClass="form-control"
                            className="input-item"
                            componentClass="input"
                            onChange={[Function]}
              />
</FormGroup>,
          ],
          "componentClass": "form",
          "horizontal": false,
          "inline": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
>
                  Person
</ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Person",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Person",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "input-person",
                  "componentClass": "input",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
>
                  Item
</ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  className="input-item"
                  componentClass="input"
                  onChange={[Function]}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Item",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Item",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "input-item",
                  "componentClass": "input",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Remove Wish",
          "className": "btn-remove",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove Wish",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={false}
>
            <FormGroup
                        bsClass="form-group"
            >
                        <ControlLabel
                                    bsClass="control-label"
                                    srOnly={false}
                        >
                                    Person
                        </ControlLabel>
                        <FormControl
                                    bsClass="form-control"
                                    className="input-person"
                                    componentClass="input"
                                    onChange={[Function]}
                        />
            </FormGroup>
            <FormGroup
                        bsClass="form-group"
            >
                        <ControlLabel
                                    bsClass="control-label"
                                    srOnly={false}
                        >
                                    Item
                        </ControlLabel>
                        <FormControl
                                    bsClass="form-control"
                                    className="input-item"
                                    componentClass="input"
                                    onChange={[Function]}
                        />
            </FormGroup>
</Form>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="btn-remove"
            disabled={false}
            onClick={[Function]}
>
            Remove Wish
</Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form",
            "children": Array [
              <FormGroup
                bsClass="form-group"
>
                <ControlLabel
                                bsClass="control-label"
                                srOnly={false}
                >
                                Person
                </ControlLabel>
                <FormControl
                                bsClass="form-control"
                                className="input-person"
                                componentClass="input"
                                onChange={[Function]}
                />
</FormGroup>,
              <FormGroup
                bsClass="form-group"
>
                <ControlLabel
                                bsClass="control-label"
                                srOnly={false}
                >
                                Item
                </ControlLabel>
                <FormControl
                                bsClass="form-control"
                                className="input-item"
                                componentClass="input"
                                onChange={[Function]}
                />
</FormGroup>,
            ],
            "componentClass": "form",
            "horizontal": false,
            "inline": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
>
                    Person
</ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    className="input-person"
                    componentClass="input"
                    onChange={[Function]}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Person",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Person",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "className": "input-person",
                    "componentClass": "input",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
>
                    Item
</ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    className="input-item"
                    componentClass="input"
                    onChange={[Function]}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Item",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Item",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "className": "input-item",
                    "componentClass": "input",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Remove Wish",
            "className": "btn-remove",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove Wish",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
